src/twisted/protocols/basic.py

```diff
@@ -419,7 +422,7 @@ class LineOnlyReceiver(protocol.Protocol):
 
     _buffer = b""
     delimiter = b"\r\n"
-    MAX_LENGTH = 16384
+    MAX_LENGTH = _MAX_LINE_LENGTH
```

Comment:
Making this as a change to *all* `LineOnlyReceiver` and `LineReceiver` uses seems pretty excessive.  The issue is with certain HTTP servers.  Assuming the right fix is to make Twisted more liberal in what it accepts from HTTP servers, this doesn't mean *all* line-based protocols implemented with Twisted should have the same change applied.

I'd move this fix into the HTTP implementation and leave `LineReceiver` / `LineOnlyReceiver` alone.

Reply:
Agreed. I have now limited the scope of the changes to `HTTPClientParser`.